### ISSUE TITLE
IDE0027 ignores line-wrapped array return in accessor expression-body check

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#84395]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/IDE0027-ignores-line-wrapped-array-retur/11112724)._

---
[severity:It's more difficult to complete my work]
Use expression body for accessors (IDE0027) rule doesnt work correctly. With below configuration in .editorconfig provided example of two properties which are logically and semantically same they will produce different output for IDE0027 only because of line wrapping. First property will trigger IDE0027, second will not. Behavior should be same for both cases.

csharp_style_expression_bodied_properties = false

dotnet_diagnostic.IDE0025.severity = error #Use expression body for properties (Prefer block)
dotnet_diagnostic.IDE0027.severity = error #Use expression body for accessors (when_on_single_line)

```c#
        public PatientDataInfo[] DatasetsToImport1
        {
            // this is triggering IDE0027
            get
            {
                return new PatientDataInfo[] {};
            }
        }

        public PatientDataInfo[] DatasetsToImport2
        {
            // this is not triggering IDE0027 and it should
            get
            {
                return new PatientDataInfo[]
                {
                };
            }
        }
```

---
#### Feedback Bot on 6/25/2026, 09:22 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

**Comment by CyrusNajmabadi** [comment]
> First property will trigger IDE0027, second will not. 

THis is exactly the point of "when on single line".  It's to say if you're ok using an expression bodied member if the body is multiple lines.  Since:

```
                return new PatientDataInfo[]
                {
                };
```

Is 3 lines, and you've said you only want expression bodies when on a single line... it is not offered.  That's teh point of the option.
